polaris-service/src/test/java/org/apache/polaris/service/admin/PolarisOverlappingTableTest.java

```diff
@@ -0,0 +1,304 @@
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+@ExtendWith({DropwizardExtensionsSupport.class, PolarisConnectionExtension.class})
+public class PolarisOverlappingTableTest {
+  private static final DropwizardAppExtension<PolarisApplicationConfig> BASE_EXT =
+      new DropwizardAppExtension<>(
+          PolarisApplication.class,
+          ResourceHelpers.resourceFilePath("polaris-server-integrationtest.yml"),
+          // Bind to random port to support parallelism
+          ConfigOverride.config("server.applicationConnectors[0].port", "0"),
+          ConfigOverride.config("server.adminConnectors[0].port", "0"),
+          // Enforce table location constraints
+          ConfigOverride.config("featureConfiguration.ALLOW_UNSTRUCTURED_TABLE_LOCATION", "false"),
+          ConfigOverride.config("featureConfiguration.ALLOW_TABLE_LOCATION_OVERLAP", "false"));
+
+  private static final DropwizardAppExtension<PolarisApplicationConfig> LAX_EXT =
+      new DropwizardAppExtension<>(
+          PolarisApplication.class,
+          ResourceHelpers.resourceFilePath("polaris-server-integrationtest.yml"),
+          // Bind to random port to support parallelism
+          ConfigOverride.config("server.applicationConnectors[0].port", "0"),
+          ConfigOverride.config("server.adminConnectors[0].port", "0"),
+          // Relax table location constraints
+          ConfigOverride.config("featureConfiguration.ALLOW_UNSTRUCTURED_TABLE_LOCATION", "true"),
+          ConfigOverride.config("featureConfiguration.ALLOW_TABLE_LOCATION_OVERLAP", "true"));
+
+  private static PolarisConnectionExtension.PolarisToken adminToken;
+  private static String userToken;
+  private static String realm;
+  private static String namespace;
+  private static final String baseLocation = "file:///tmp/PolarisOverlappingTableTest";
+
+  private static final CatalogWrapper defaultCatalog = new CatalogWrapper("default");
+  private static final CatalogWrapper laxCatalog = new CatalogWrapper("lax");
+  private static final CatalogWrapper strictCatalog = new CatalogWrapper("strict");
+
+  /** Used to define a parameterized test config */
+  protected record TestConfig(
+      DropwizardAppExtension<PolarisApplicationConfig> extension,
+      CatalogWrapper catalogWrapper,
+      Response.Status response) {
+    public String catalog() {
+      return catalogWrapper.catalog;
+    }
+
+    private String extensionName() {
+      return (extension
+              .getConfiguration()
+              .getConfigurationStore()
+              .getConfiguration(null, PolarisConfiguration.ALLOW_TABLE_LOCATION_OVERLAP))
+          ? "lax"
+          : "strict";
+    }
+
+    /** Extract the first component of the catalog name; e.g. `default` from `default_123_xyz` */
+    private String catalogShortName() {
+      StringBuilder result = new StringBuilder();
```

Comment:
Doesn't really matter but, 
```java
firstComponentEnd = catalog().indexOf('_')

if (firstComponentEnd != -1) {
  return catalog().substring(0, firstComponentEnd)
} else {
  return catalog()
}
```

Would avoid the control loop and allocation of the StringBuilder and take advantage of the String interning already happening

Reply:
Thanks, this is great. 

I still miss Scala: `string.toCharArray.takeWhile(c => c != '_').mkString`

Reply:
I know, I can tell you want to write Scala and Java is breaking your heart. It breaks mine every day